tests/run-make/core-ffi-typecheck/rmake.rs

```diff
@@ -0,0 +1,168 @@
+use std::collections::HashMap;
+
+use run_make_support::{clang, rfs, rustc, serde_json};
+
+const C_TYPES: &[(&str, &str)] = &[
+    ("c_char", "__CHAR_BIT__"), // Character width
+    ("char_signedness", "__CHAR_UNSIGNED__"),
+    ("c_double", "__SIZEOF_DOUBLE__"), // Double precision floating-point
+    ("c_float", "__SIZEOF_FLOAT__"),   // Single precision floating-point
+    ("c_int", "__SIZEOF_INT__"),       // Signed integer
+    ("c_long", "__SIZEOF_LONG__"),     // Signed long integer
+    ("c_longlong", "__SIZEOF_LONG_LONG__"), // Signed long long integer
+    ("c_schar", "__SIZEOF_CHAR__"),    // Signed char
+    ("c_short", "__SIZEOF_SHORT__"),   // Signed short integer
+    ("c_uchar", "__SIZEOF_CHAR__"),    // Unsigned char
+    ("c_uint", "__SIZEOF_INT__"),      // Unsigned integer
+    ("c_ulong", "__SIZEOF_LONG__"),    // Unsigned long integer
+    ("c_ulonglong", "__SIZEOF_LONG_LONG__"), // Unsigned long long integer
+    ("c_ushort", "__SIZEOF_SHORT__"),  // Unsigned short integer
+    ("c_size_t", "__SIZEOF_SIZE_T__"), // Size type
+    ("c_ptrdiff_t", "__SIZEOF_PTRDIFF_T__"), // Pointer difference type];
```

Comment:
We should either prefer the standard macros of `CHAR_WIDTH` and so on where available (which will be in bits instead of bytes, but eh, just hit it with `>> 3`), or simply use the actual `sizeof` operator within the generated C code.